__tests__/schema/convert-edtr-io-state.ts

```diff
@@ -0,0 +1,872 @@
+//@ts-nocheck
+
+import { convert } from '@/schema/convert-edtr-io-state'
+
+function log(result) {
+  console.log('-------------------------')
+  console.log(JSON.stringify(result))
+  console.log('-------------------------')
+}
+
+describe('general compat cases / sanity checks', () => {
+  test('undefined node, return empty array', () => {
+    const result = convert(undefined)
+    expect(result).toEqual([])
+  })
+
+  test('node is empty object, return empty array', () => {
+    const result = convert({})
+    expect(result).toEqual([])
+  })
+
+  test('unsupported plugin, return empty array', () => {
+    const result = convert({ plugin: 'woah-nice-plugin', state: [] })
+    expect(result).toEqual([])
+  })
+
+  test('unsupported text type, return empty array', () => {
+    const result = convert({ type: 'super-bold', children: [] })
+    expect(result).toEqual([])
+  })
+})
+
+describe('edtr io plugins', () => {
+  // plugin: files currently not supported in editor or frontend
+  // plugin: inputExercise not handled in converter
+  // plugin: scMcExercise not handled in converter
+
+  //TODO: Add test for Equations
+
+  describe('plugin: image', () => {
+    test('default', () => {
+      const result = convert({
+        plugin: 'image',
+        state: {
+          src:
+            'https://assets.serlo.org/5c77cd8b27d83_b56b69f307447a110a5ae915e517c73e385c37e8.jpg',
+          alt: 'Lernen im eigenen Tempo mit serlo.org',
+        },
+      })
+      expect(result).toEqual([
+        {
+          type: 'img',
+          src:
+            'https://assets.serlo.org/5c77cd8b27d83_b56b69f307447a110a5ae915e517c73e385c37e8.jpg',
+          alt: 'Lernen im eigenen Tempo mit serlo.org',
+          maxWidth: undefined,
+        },
+      ])
+    })
+    test('with maxWidth', () => {
+      const result = convert({
+        plugin: 'image',
+        state: {
+          src:
+            'https://assets.serlo.org/5c77cd8b27d83_b56b69f307447a110a5ae915e517c73e385c37e8.jpg',
+          alt: 'Lernen im eigenen Tempo mit serlo.org',
+          maxWidth: true,
+        },
+      })
+      expect(result).toEqual([
+        {
+          type: 'img',
+          src:
+            'https://assets.serlo.org/5c77cd8b27d83_b56b69f307447a110a5ae915e517c73e385c37e8.jpg',
+          alt: 'Lernen im eigenen Tempo mit serlo.org',
+          maxWidth: true,
+        },
```

Comment:
```suggestion
        expect.objectContaining({ maxWidth: true }),
```

An example when you want emphase that `maxWidth` is passed to the result

Reply:
anything against `expect(result.maxWidth).toBe(true)`? It's a bit easier to read for me.

Reply:
`expect(result).toEqual([ expect.objectContaining({ maxWidth: true }) ])`